cmd/kubectl-directpv/init_config.go

```diff
@@ -0,0 +1,84 @@
+// This file is part of MinIO DirectPV
+// Copyright (c) 2021, 2022 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	initConfigVersionV1 = "v1"
+)
+
+// InitConfigV1 defines the config to initialize the devices
+type InitConfigV1 struct {
+	Version string     `yaml:"version"`
+	Nodes   []NodeInfo `yaml:"nodes,omitempty"`
+}
+
+// NodeInfo holds the node information
+type NodeInfo struct {
+	Name   string      `yaml:"name"`
+	Drives []DriveInfo `yaml:"drives,omitempty"`
+}
+
+// DriveInfo represents the drives that are to be initialized
+type DriveInfo struct {
+	ID         string `yaml:"id"`
+	Name       string `yaml:"name"`
+	MajorMinor string `yaml:"majorMinor"`
+	FS         string `yaml:"fs,omitempty"`
+}
+
+func writeInitConfig(config InitConfigV1) error {
+	f, err := os.OpenFile(outputFile, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0o644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	config.Version = initConfigVersionV1
+	b, err := yaml.Marshal(config)
+	if err != nil {
+		return err
+	}
+	if _, err = f.Write(b); err != nil {
+		return err
+	}
+	return nil
+}
+
+func readInitConfig(inputFile string) (config InitConfigV1, err error) {
+	var data []byte
+	data, err = ioutil.ReadFile(inputFile)
```

Comment:
Use https://pkg.go.dev/gopkg.in/yaml.v2#NewDecoder instead of reading whole file.